src/scene/particle-container/shared/Particle.ts

```diff
@@ -0,0 +1,167 @@
+import type { ColorSource } from '../../../color/Color';
+
+/**
+ * Represents a particle with properties for position, scale, rotation, color, and texture.
+ * @property {number} x - The x-coordinate of the particle.
+ * @property {number} y - The y-coordinate of the particle.
+ * @property {number} scaleX - The scale factor in the x-axis.
+ * @property {number} scaleY - The scale factor in the y-axis.
+ * @property {number} anchorX - The x-coordinate of the anchor point.
+ * @property {number} anchorY - The y-coordinate of the anchor point.
+ * @property {number} rotation - The rotation of the particle in radians.
+ * @property {number} color - The color of the particle as a hexadecimal number.
+ * @property {Texture} texture - The texture of the particle.
+ * @memberof scene
+ */
+export interface IParticle
+{
+    x: number;
+    y: number;
+    scaleX: number;
+    scaleY: number;
+    anchorX: number;
+    anchorY: number;
+    rotation: number;
+    color: number;
+    texture: Texture;
+}
+
+/**
+ * Represents the options for creating a new particle.
+ * @property {number} x - The x-coordinate of the particle.
+ * @property {number} y - The y-coordinate of the particle.
+ * @property {number} scaleX - The scale factor in the x-axis.
+ * @property {number} scaleY - The scale factor in the y-axis.
+ * @property {number} anchorX - The x-coordinate of the anchor point.
+ * @property {number} anchorY - The y-coordinate of the anchor point.
+ * @property {number} rotation - The rotation of the particle in radians.
+ * @property {Texture} texture - The texture of the particle.
+ * @property {ColorSource} tint - The tint color of the particle as a hexadecimal number.
+ * @property {number} alpha - The alpha value of the particle.
+ * @memberof scene
+ */
+export type ParticleOptions = Omit<Partial<IParticle>, 'color'> & {
+    texture: Texture
+    tint?: ColorSource;
+    alpha?: number;
+};
+
+/**
+ * Represents a single particle within a particle container. This class implements the IParticle interface,
+ * providing properties and methods to manage the particle's position, scale, rotation, color, and texture.
+ *
+ * The reason we use a particle over a sprite is that these are much lighter weight and we can create a lot of them
+ * without taking on the overhead of a full sprite.
+ *
+ * Here is an example of how to create a new particle:
+ *
+ * ```javascript
+ * const particle = new Particle({
+ *   texture,
+ *   x: 100,
+ *   y: 100,
+ *   scaleX: 0.5,
+ *   scaleY: 0.5,
+ *   rotation: Math.PI / 2,
+ *   color: 0xff0000,
+ * });
+ * ```
+ * @implements {IParticle}
+ * @memberof scene
+ */
+export class Particle implements IParticle
+{
+    /** Default options for constructing with options */
+    public static defaultOptions: Partial<ParticleOptions> = {
+        anchorX: 0,
+        anchorY: 0,
+        x: 0,
+        y: 0,
+        scaleX: 1,
+        scaleY: 1,
+        rotation: 0,
+        tint: 0xffffff,
+        alpha: 1,
+    };
```

Comment:
@GoodBoyDigital i tested bunny mark with using particles and get ~52 fps with 1 million particles which is identical to not using a default options object

Reply:
woop! great new :D